### PR TITLE
[MOS-45] Invalid number decoding

### PR DESCRIPTION
**Pull request by niepiekm**
Added UT case for failing phone number
Fixed decimal semi-octet decoding (AKA ascii combination)

### CODE REVIEW

File: module-utils/unicode/utf8/test/unittest_utf8.cpp [hunk 1]
```diff
@@ -356,6 +356,13 @@ TEST_CASE("UTF8: Not ASCII combination")
         REQUIRE_FALSE(combination.toASCII().has_value());
     }
 
+    SECTION("Carier info number - 3 digits")
```
Comment (mjan-dev) [inline module-utils/unicode/utf8/test/unittest_utf8.cpp:359]: Could you please add more UTs? For example with other 3/2/1 digits combinations.

Comment (niepiekm) [reply module-utils/unicode/utf8/test/unittest_utf8.cpp:359]: Working on it.

File: module-utils/unicode/utf8/test/unittest_utf8.cpp [hunk 2]
```diff
@@ -356,6 +356,13 @@ TEST_CASE("UTF8: Not ASCII combination")
...
+    {
+        UTF8 combination = "100";
```
Comment (pholat) [inline module-utils/unicode/utf8/test/unittest_utf8.cpp:361]: please add tests for at least: 
for negative cases:
- 1111  
- 11111 
- 111111
positive cases
- 11147
- 8411
- 11182115

File: module-utils/unicode/utf8/utf8/UTF8.cpp [hunk 3]
```diff
@@ -798,7 +798,7 @@ bool UTF8::isASCIICombination() const noexcept
     constexpr uint8_t secondCharacterFactor = 10;
     for (; i < len; i += 2) {
         int firstCharacter = 0;
-        if (data[i] == '1') {
+        if (len > 3 && data[i] == '1') {
```
Comment (niepiekm) [inline module-utils/unicode/utf8/utf8/UTF8.cpp:801]: Considering what I found out from the spec and observation of Quectel eval board, this seems to be just a workaround this particular problem with `100` number. I'm yet to figure out how to methodically distinguish a number consisting from digits from alphanumeric encoded using BSD.

File: module-utils/unicode/utf8/test/unittest_utf8.cpp (hunk 1 above)
Comment (mjan-dev) [reply module-utils/unicode/utf8/test/unittest_utf8.cpp]: Ok, I see added test cases, thanks.

File: module-utils/unicode/utf8/utf8/UTF8.cpp (hunk 3 above)
Comment (mjan-dev) [reply module-utils/unicode/utf8/utf8/UTF8.cpp:801]: Please add comment clearly stating that it is a workaround for this particular case.

Comment (niepiekm) [reply module-utils/unicode/utf8/utf8/UTF8.cpp]: Done

File: module-utils/unicode/utf8/test/unittest_utf8.cpp (hunk 2 above)
Comment (niepiekm) [reply module-utils/unicode/utf8/test/unittest_utf8.cpp:398]: Added representative TCs

**Review by mjan-dev — APPROVED** [review]
Approving as a workaround for this particular case, not a systematic solution to the problem.

**Review by kkleczkowski — APPROVED** [review]

**Pull request merged**